src/foam/nanos/controller/ApplicationController.js

```diff
@@ -535,7 +536,7 @@ foam.CLASS({
           // only push the unauthenticated menu if there is no subject
           // if client is authenticated, go on to fetch theme and set loginsuccess before pushing menu
           // use the route instead of the menu so that the menu could be re-created under the updated context
-          this.routeTo(menu.id);
+          route_initialized ? this.routeTo(menu.id) : this.pushMenu(menu);
```

Comment:
Not sure why we cant always routeTo?

Reply:
routeTo only changes the hash but if the menu.id is already the current hash it won't have any effect.